frontend/src/app/components/channel/channel.controller.ts

```diff
@@ -80,14 +87,15 @@ export class BaseChannelController {
 
     // TODO: The timestamp should be better taken from Janus point of view
     user.joined = moment();
-    user.disabled = false;
+    user.disabled = this.publisherStatus.connectStatusByLogin(login) === InternetConnectionType.danger;
```

Comment:
At any place where you use this.publisherStatus.connectStatusByLogin(login) update user.connectionStatus and then use user.connectionStatus in all required places.